common/ethereum/transaction.py

```diff
@@ -97,10 +93,10 @@ def has_chain_id(self) -> bool:
 
     @cached_property
     def chain_id(self) -> int | None:
-        return self.calc_chain_id(self.v)
+        return self._calc_chain_id(self.v)
 
     @staticmethod
-    def calc_chain_id(v: int) -> int | None:
+    def _calc_chain_id(v: int) -> int | None:
```

Comment:
There are two places:
- indexer/db/neon_tx_db.py:268 - neon_tx = NeonTxModel(...)
- indexer/base/objects.py:414 - return NeonTxModel(...)

In both places, there is no `chain-id` -property, but they have the `v`-property.
And you removed the cached-property `chain-id` from the NeonTxModel:
```
@cached_property		
def chain_id(self) -> int | None:		
    return EthTx.calc_chain_id(self.v)
```

So, before your modifications, the Proxy loaded NeonTxModel without `chain-id`, but with `v`.
And then calculated `chain-id` from the `v`.

Storing Tx as a blob-RLP requires additional steps because converting the old db to the new - is a very long process of about 0.5 hours.

So, we will have a grace period, when we should support 2 modes:
- if a record doesn't have a blob-RLP - load just `v` and convert it to `chain-id` for legacy txs
- if the record has a blob-RLP - loading blob-RLP and extracting all tx-properties from it

Or another variant:
- if the record doesn't have a blob-RLP - load parsed Tx, pack into RLP, and extract the result from RLP ))
- if the record has a blob-RLP - just load blob-RLP and extract all tx-properties from it

The second variant looks better because we can use the same logic for extracting tx-properties. And after converting DB in the background, we can remove the legacy code.

Reply:
Thank you!

I'll go with the second option.

Reply:
@afalaleev do I understand correctly, that the plan is to have an additional blob column (let's call it "tx_rlp") and **leave other columns in place**? 
This way obviously our table will grow ~2 times, but the benefit is just convenient construction of `NeonTxMetaModel` in the code. 

Could you please confirm I understood it correctly?

Reply:
+CC @spylogsster